cmd/notify-mailer/main.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	netmail "net/mail"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/mail"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+type mailer struct {
+	clk           clock.Clock
+	log           blog.Logger
+	dbMap         *gorp.DbMap
+	mailer        mail.Mailer
+	subject       string
+	emailTemplate string
+	destinations  []string
+	checkpoint    interval
+	sleepInterval time.Duration
+}
+
+type interval struct {
+	start int
+	end   int
+}
+
+func (i *interval) isSane() error {
```

Comment:
This is a bit of a language nit, but I somewhat regretted using "IsSane" for challenges due to its unnecessary mental health connotations. Want to just call this method `ok()` ?